cli/commands/play.go

```diff
@@ -109,26 +122,50 @@ var run = func(cmd *cobra.Command, args []string) {
 		if TurnDelay > 0 {
 			time.Sleep(time.Duration(TurnDelay) * time.Millisecond)
 		}
+
+		if exportGame {
+			// The output file was designed in a way so that (nearly) every entry is equivalent to a valid API request.
+			// This is meant to help unlock further development of tools such as replaying a saved game by simply copying each line and sending it as a POST request.
+			// There was a design choice to be made here: the difference between SnakeResponse and BoardState is the `you` key.
+			// We could choose to either store the SnakeResponse of each snake OR to omit the `you` key OR fill the `you` key with one of the snakes
+			// In all cases the API request is technically non-compliant with how the actual API response should be.
+			// The third option (filling the `you` key with an arbitrary snake) is the closest to the actual API response that would need the least manipulation to
+			// be adjusted to look like an API call for a specific snake in the game.
+			snakeState := snakeStates[state.Snakes[0].ID]
+			snakeRequest := getIndividualBoardStateForSnake(state, snakeState, snakeStates, ruleset)
+			gameExporter.AddSnakeRequest(snakeRequest)
+		}
 	}
 
+	isDraw := true
 	if GameType == "solo" {
 		log.Printf("[DONE]: Game completed after %v turns.", Turn)
+		if exportGame {
+			gameExporter.winner = snakeStates[state.Snakes[0].ID]
+		}
 	} else {
-		var winner string
-		isDraw := true
+		var winner SnakeState
 		for _, snake := range state.Snakes {
+			snakeState := snakeStates[snake.ID]
 			if snake.EliminatedCause == rules.NotEliminated {
 				isDraw = false
-				winner = snakeStates[snake.ID].Name
+				winner = snakeState
 			}
-			sendEndRequest(ruleset, state, snakeStates[snake.ID], snakeStates)
+			sendEndRequest(ruleset, state, snakeState, snakeStates)
 		}
 
 		if isDraw {
 			log.Printf("[DONE]: Game completed after %v turns. It was a draw.", Turn)
 		} else {
-			log.Printf("[DONE]: Game completed after %v turns. %v is the winner.", Turn, winner)
+			log.Printf("[DONE]: Game completed after %v turns. %v is the winner.", Turn, winner.Name)
 		}
+		if exportGame {
+			gameExporter.winner = winner
+		}
+	}
+
+	if exportGame {
+		gameExporter.FlushToFile(Output, "JSONL")
```

Comment:
The error returned here isn't being checked. I think we'd want to log and exit with an error if it's not nil? Otherwise a bunch of games run by a script could end up without files.

Reply:
Added a log and an exit with status code 1 for this.